development/workflow.rst

```diff
@@ -0,0 +1,501 @@
+###############################
+LSST Stack Development Workflow
+###############################
+
+This page describes a complete workflow for developing the LSST Stack.
+
+1. :ref:`lsstsw-workflow-obtaining-lsstsw-stack`
+2. :ref:`lsstsw-workflow-working-ticket`
+3. :ref:`lsstsw-workflow-jenkins`
+4. :ref:`lsstsw-workflow-pr`
+5. :ref:`lsstsw-workflow-rebuild`
+6. :ref:`lsstsw-workflow-ext`
+
+.. TODO once more workflows are published, link to them here as well
+
+.. _lsstsw-workflow-prerequisites:
+
+Prerequisites
+=============
+
+Before developing the Stack, ensure you have all software dependencies installed on your system.
+These dependencies are listed in the :doc:`guide to installing the stack from source </install>`.
+
+Ensure that your git author information reflects your work email address.
+
+.. prompt:: bash
+
+   git config --global user.name
+   git config --global user.email
+
+Set them if necessary:
+
+.. prompt:: bash
+
+   git config --global user.name  {{First Last}}
+   git config --global user.email {{your official LSST email}}
+
+.. TODO include git lfs installation instructions? perhaps link this info to an in-depth git guide
+
+.. _lsstsw-workflow-obtaining-lsstsw-stack:
+
+Obtaining a Development Stack with lsstsw
+=========================================
+
+Code for the LSST Stack is distributed across many git repositories (see `github.com/lsst <https://github.com/lsst>`_).
+`lsstsw <https://github.com/lsst/lsstsw>`_ is a tool that helps you manage the codebase by automating the process of cloning all of these repositories and building that development Stack for testing.
+
+.. _lsstsw-workflow-obtaining-lsstw-stack-get:
+
+Step 1. Get lsstsw
+------------------
+
+Begin by choosing a work directory, then clone ``lsstsw`` into it:
+
+.. prompt:: bash
+
+   git clone git@github.com:lsst/lsstsw.git
+   cd lsstsw
+
+You may want to edit the file ``etc/repos.yaml`` to use SSH authentication so that pushing changes is easier.
+This can be done with ``sed`` in the ``lsstw/`` directory:
+
+.. prompt:: bash
+
+   sed -i '' 's/https:\/\/github.com\/lsst/git@github.com:lsst/g' etc/repos.yaml
+
+.. NOTE that this will break a future git-lfs based afwdata that relies on HTTPS transport.
```

Comment:
I don't think it breaks afwdata since the GitHub repo still can use ssh; isn't it the S3 back-end that needs HTTPS?